### PR TITLE
fixes: PWA & interface on small screens

### LINKED ISSUE #2364: [v1.0.0b] - "app" icon colors
### First Check

- [X] This is not a feature request
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.

### What is the issue you are experiencing?

I've found that if I save my Mealie page as an app I get an icon with an orange knife and fork on a orange background. That's using Samsung's default internet browser.
But, if I do the exact same thing using Chrome on the same phone I get an orange knife and fork on a white background.
See attached images.
My phone is a Samsung Galaxy S22.  And I'm running omni-nightly.
Screenshot_20230427_194926_One_UI_Home
Screenshot_20230427_132950_One_UI_Home

### Deployment

Docker (Linux)

### Deployment Details

_No response_

### PR DESCRIPTION
**Pull request by p0lycarpio**
## What type of PR is this?

_(REQUIRED)_

- bug
- cleanup

## What this PR does / why we need it:

_(REQUIRED)_

This PR improves the PWA icons used on smartphone and disable the default icon generation with nuxt/pwa.
This also improves the readibility of actions buttons and fix spacing on small screens.

i18n : missing strings are added

## Which issue(s) this PR fixes:

_(REQUIRED)_

Fixes #2364

## Release Notes

_(REQUIRED)_

```release-note
improve PWA icons & UI on small screens
regenerate translation strings
```

### CODE REVIEW

**Comment by hay-kot** [comment]
Can you post some screenshots of these UI changes?

**Comment by p0lycarpio** [comment]
Here are some examples of interface changes on small screens.

**Before / After**

## Login page
 

## Recipe editing
 

## User settings
same in group and other pages
 

---
## PWA icons changes
warning and cropped maskable icons

**Before**
Capture d’écran du 2023-09-14 18-07-57
Capture d’écran du 2023-09-14 18-08-11

**After**
Capture d’écran du 2023-09-14 18-08-21
Capture d’écran du 2023-09-14 18-08-36

**Comment by p0lycarpio** [comment]
Here are some examples of interface changes on small screens.

**Before / After**

## Login page
 

## Recipe editing
 

## User settings
same in group and other pages
 

---
## PWA icons changes
warning and cropped maskable icons

**Before**
Capture d’écran du 2023-09-14 18-07-57
Capture d’écran du 2023-09-14 18-08-11

**After**
Capture d’écran du 2023-09-14 18-08-21
Capture d’écran du 2023-09-14 18-08-36

**Comment by hay-kot** [comment]
These changes are lovely thank you!

File: frontend/pages/admin/backups.vue [hunk 1]
```diff
@@ -108,8 +108,8 @@
         </div>
       </section>
     </section>
-    <v-container class="mt-4 d-flex justify-end">
-      <v-btn outlined rounded to="/group/migrations"> {{ $t('recipe.looking-for-migrations') }} </v-btn>
+    <v-container class="mt-4 d-flex justify-center text-center">
+      <a to="/group/migrations"> {{ $t('recipe.looking-for-migrations') }} </a>
```
Comment (hay-kot) [inline frontend/pages/admin/backups.vue]: Why did you change this to an `a` tag? I believe it should be a Nuxt Link https://nuxtjs.ir/api/components-nuxt-link

File: frontend/pages/group/members.vue [hunk 2]
```diff
@@ -13,10 +13,10 @@
             <b>{{ $t('group.invite') }}</b>
           </template>
         </i18n>
-    </BasePageTitle>
-    <v-container class="mt-4 d-flex justify-start">
-      <v-btn outlined rounded to="/user/profile/edit"> {{ $t('group.looking-to-update-your-profile') }} </v-btn>
-    </v-container>
+        <v-container class="mt-1 px-0">
+        <a class="text-center" to="/user/profile/edit"> {{ $t('group.looking-to-update-your-profile') }} </a>
```
Comment (hay-kot) [inline frontend/pages/group/members.vue]: Why did you change this to an a tag? I believe it should be a Nuxt Link https://nuxtjs.ir/api/components-nuxt-link

File: frontend/yarn.lock [hunk 3]
```diff
@@ -1910,18 +1910,20 @@
...
-    fs-extra "^9.0.1"
+"@nuxtjs/pwa@^3.3.5":
+  version "3.3.5"
+  resolved "https://registry.yarnpkg.com/@nuxtjs/pwa/-/pwa-3.3.5.tgz#db7c905536ebe8a464a347b6ae3215810642c044"
+  integrity sha512-8tTmW8DBspWxlJwTimOHTkwfkwPpL9wIcGmy75Gcmin+c9YtX2Ehxmhgt/TLFOC9XsLAqojqynw3/Agr/9OE1w==
+  dependencies:
+    clone-deep "^4.0.1"
+    defu "^3.2.2"
+    execa "^5.0.0"
+    fs-extra "^9.1.0"
```
Comment (hay-kot) [inline frontend/yarn.lock]: We can't upgrade this dependency. It will break the PWA

- https://github.com/mealie-recipes/mealie/issues/1077
- https://github.com/nuxt-community/pwa-module/issues/528

File: frontend/pages/user/profile/edit.vue [hunk 4]
```diff
@@ -108,14 +108,14 @@
         :label="$t('profile.show-advanced-description')"
         @change="updateUser"
       ></v-checkbox>
+      <a class="mt-5 d-flex flex-column justify-center text-center" to="/group"> {{ $t('profile.looking-for-privacy-settings') }} </a>
```
Comment (hay-kot) [inline frontend/pages/user/profile/edit.vue]: Why did you change this to an a tag? I believe it should be a Nuxt Link https://nuxtjs.ir/api/components-nuxt-link

**Review by hay-kot — CHANGES_REQUESTED** [review]
Couple changes need to be made, but overall looks good.

For the translations. You should only modify the english set of json files. The rest are auto-generated and PR is made by an external service, so those need to be changed as well.

**Comment by p0lycarpio** [comment]
Thank you for your feedback ! I made the modifications. For translations, I know that Mealie uses Crowdwin but the English-only strings are not present in the other languages interface. Is there a pipeline to launch?

**Comment by hay-kot** [comment]
> Thank you for your feedback ! I made the modifications. For translations, I know that Mealie uses Crowdwin but the English-only strings are not present in the other languages interface. Is there a pipeline to launch?

Thanks! The other languages should fallback if there's no reference. But once this is merged Crowdin will automatically fill those with defaults and then people can start translating them. I realized that the Crowdin integration has been broken for some time, so I wanted to get that fixed before I merged this. Thanks for your patience!

Should be fixed today and then I'll merge this!
